drivers/adc/ad7616/ad7616.c

```diff
@@ -369,24 +604,33 @@ int32_t ad7616_set_oversampling_ratio(struct ad7616_dev *dev,
 					AD7616_OS(0x7), AD7616_OS(osr));
 	} else {
 		ret = no_os_gpio_set_value(dev->gpio_os0, ((osr & 0x01) >> 0));
-		ret |= no_os_gpio_set_value(dev->gpio_os1, ((osr & 0x02) >> 1));
-		ret |= no_os_gpio_set_value(dev->gpio_os2, ((osr & 0x04) >> 2));
+		if (ret != 0)
+			return ret;
+
+		ret = no_os_gpio_set_value(dev->gpio_os1, ((osr & 0x02) >> 1));
+		if (ret != 0)
+			return ret;
+
+		ret = no_os_gpio_set_value(dev->gpio_os2, ((osr & 0x04) >> 2));
+		if (ret != 0)
+			return ret;
```

Comment:
this change is not related to the commit message. 
either you leave it as it was, or you create a new commit in which you replace ret |= with a separate call each time it is used (currently there is another instance in which this logic is used but is kept as it is, see in ad7616_set_range). 
see all instances in which you performed this change: e.g. in ad7616_set_mode